src/composed/shared.rs

```diff
@@ -134,3 +170,17 @@ pub(crate) fn filter_parsed_packet_results(p: Result<Packet>) -> Option<Result<P
         }
     }
 }
+
+/// Check if the OpenPGP data in `input` seems to be ASCII-armored or binary (by looking at the
+/// highest bit of the first byte)
+pub(crate) fn is_binary<R: Read + Seek>(input: &mut R) -> Result<bool> {
+    // Peek at the first byte in the reader
+    let first = input.read_u8()?;
+    let _ = input.seek(SeekFrom::Start(0))?;
```

Comment:
This is not correct, we don't know if the source is at `0`. We should use `stream_position` before reading to store the current position

Reply:
or just use `SeekFrom::Current(-1)`

Reply:
good catch. i changed this to the relative-seek approach.